### PR TITLE
[Fusion] Added pre-merge validation rule "LookupMustNotReturnListRule"

### PR DESCRIPTION
**Pull request by danielreynolds1**
Summary of the changes (Less than 80 chars)

- [Fusion] Added pre-merge validation rule "LookupMustNotReturnListRule"

### CODE REVIEW

File: src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Logging/LogEntryHelper.cs [hunk 1]
```diff
@@ -257,6 +257,25 @@ public static LogEntry KeyInvalidSyntax(
...
+
+        return new LogEntry(
+            string.Format(
+                LogEntryHelper_LookupMustNotReturnList,
+                coordinate,
+                schema.Name),
+            LogEntryCodes.LookupMustNotReturnList,
+            LogSeverity.Error,
+            coordinate,
+            type,
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Logging/LogEntryHelper.cs:275]:
```suggestion
            field,
```

File: src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Properties/CompositionResources.resx [hunk 2]
```diff
@@ -63,6 +63,9 @@
   <data name="LogEntryHelper_KeyInvalidSyntax" xml:space="preserve">
     <value>A @key directive on type '{0}' in schema '{1}' contains invalid syntax in the 'fields' argument.</value>
   </data>
+  <data name="LogEntryHelper_LookupMustNotReturnList" xml:space="preserve">
+    <value>Lookup field '{0}' in schema '{1}' must not return a list.</value>
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Properties/CompositionResources.resx:67]:
```suggestion
    <value>The lookup field '{0}' in schema '{1}' must not return a list.</value>
```

File: src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PreMergeValidation/Rules/LookupMustNotReturnListRule.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+using HotChocolate.Fusion.Events;
+using HotChocolate.Skimmed;
+using static HotChocolate.Fusion.Logging.LogEntryHelper;
+
+namespace HotChocolate.Fusion.PreMergeValidation.Rules;
+
+/// <summary>
+/// Fields annotated with the <c>@lookup</c> directive are intended to retrieve a single entity
+/// based on provided arguments. To avoid ambiguity in entity resolution, such fields must return
+/// a <b>single object</b> and not a <c>list</c>. This validation rule enforces that any field
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PreMergeValidation/Rules/LookupMustNotReturnListRule.cs:10]: As mentioned, let's align the formatting with the spec, generally-speaking (use bold where the spec uses bold, etc.).

File: src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PreMergeValidation/Rules/LookupMustNotReturnListRule.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        static bool IsListType(ITypeDefinition type)
+        {
+            if (type is NonNullTypeDefinition nonNullType)
+            {
+                var innerType = nonNullType.NullableType;
+                return IsListType(innerType); // Spec mentions `IsSingleObjectType(innerType)`
+            }
+
+            return type is ListTypeDefinition;
+        }
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PreMergeValidation/Rules/LookupMustNotReturnListRule.cs:36]: I think that we can just do:

```cs
if (ValidationHelper.IsLookup(field) && field.Type.NullableType() is ListTypeDefinition)
```

Did you mention the `IsSingleObjectType` issue to Pascal?

File: src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/LookupMustNotReturnListRuleTests.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        return new TheoryData<string[]>
+        {
+            // In this example, "userById" returns a "User" object, satisfying the requirement.
+            {
+                [
+                    """
+                    type Query {
+                        userById(id: ID!): User @lookup
+                    }
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/LookupMustNotReturnListRuleTests.cs:54]:
```suggestion
                    }

```

File: src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/LookupMustNotReturnListRuleTests.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        return new TheoryData<string[], string[]>
+        {
+            // Here, "usersByIds" returns a list of "User" objects, which violates the requirement
+            // that a @lookup field must return a single object.
+            {
+                [
+                    """
+                    type Query {
+                        usersByIds(ids: [ID!]!): [User!] @lookup
+                    }
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/LookupMustNotReturnListRuleTests.cs:76]:
```suggestion
                    }

```

File: src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/LookupMustNotReturnListRuleTests.cs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                    type User {
+                        id: ID!
+                        name: String
+                    }
+                    """
+                ],
+                [
+                    "Lookup field 'Query.usersByIds' in schema 'A' must not return a list."
+                ]
+            }
```
Comment (glen-84) [inline src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/LookupMustNotReturnListRuleTests.cs:86]: We should add a test with a non-null list.

**Review by glen-84 — CHANGES_REQUESTED** [review]

**Review by glen-84 — APPROVED** [review]
